src/Plugin/views/access/GroupPermission.php

```diff
@@ -0,0 +1,234 @@
+
+  /**
+   * The group manager.
+   *
+   * @var \Drupal\og\GroupTypeManager
+   */
+  protected $groupTypeManager;
+
+  /**
+   * Constructs a GroupPermission object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\og\PermissionManagerInterface $permission_manager
+   *   The permission handler.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\og\OgAccessInterface $og_access
+   *   The OG access service.
+   * @param \Drupal\Core\Plugin\Context\ContextProviderInterface $og_context
+   *   The OG context provider.
+   * @param \Drupal\og\GroupTypeManager $group_type_manager
+   *   The group manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, PermissionManagerInterface $permission_manager, EntityTypeManagerInterface $entity_type_manager, OgAccessInterface $og_access, ContextProviderInterface $og_context, GroupTypeManager $group_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->permissionManager = $permission_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->ogAccess = $og_access;
+    $this->ogContext = $og_context;
+    $this->groupTypeManager = $group_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('og.permission_manager'),
+      $container->get('entity_type.manager'),
+      $container->get('og.access'),
+      $container->get('og.context'),
+      $container->get('og.group_type_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access(AccountInterface $account) {
+    $group = $this->getGroup();
+    if ($group) {
+      list(, $permission) = explode(':', $this->options['perm']);
+      return $this->ogAccess->userAccess($group, $permission, $account);
+    }
+    return AccessResult::neutral();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function alterRouteDefinition(Route $route) {
+    list($entity_type_id, $permission) = explode(':', $this->options['perm']);
+    $route->setRequirement('_og_user_access_group', $permission);
+    $route->setOption('_og_entity_type_id', $entity_type_id);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summaryTitle() {
+    $permissions = OptGroup::flattenOptions($this->getGroupPermissionOptions());
+    if (isset($permissions[$this->options['perm']])) {
+      return $permissions[$this->options['perm']];
+    }
+    return $this->options['perm'];
+  }
+
```

Comment:
Implemented defineOptions() and create a default for 'perm':

```suggestion

  /**
   * {@inheritdoc}
   */
  protected function defineOptions() {
    $options = parent::defineOptions();
    $options['perm'] = ['default' => ''];

    return $options;
  }

```